sdk/core/azure-core-amqp/src/impl/rust_amqp/amqp/connection.cpp

```diff
@@ -209,62 +209,70 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
     m_connectionOptions.reset(amqpconnectionoptionsbuilder_build(builder.get()));
   }
 
-  void ConnectionImpl::Open()
+  void ConnectionImpl::Open(Azure::Core::Context const& context)
   {
+    Azure::Core::Amqp::Common::_detail::CallContext callContext{
+        Azure::Core::Amqp::Common::_detail::GlobalStateHolder::GlobalStateInstance()
+            ->GetRuntimeContext(),
+        context};
     if (m_options.EnableTrace)
     {
       Log::Stream(Logger::Level::Verbose)
           << "ConnectionImpl::Open: " << this << " ID: " << m_containerId;
     }
     if (amqpconnection_open(
```

Comment:
You return `true` (or truthy) when an error occurs? Interesting. Took a couple times of re-reading this and `Close` below making sure I didn't miss some context despite GitHub showing no line gaps in the diff.

Reply:
Yeah, that's a convention that uAMQP has - it returns the line number of the error on failure and 0 on success.

I'm keeping that convention throughout the interop layer because the model types have the exact same API shape as the uAMQP model types.